src/hoc/ClientSideStateUpdate.tsx

```diff
@@ -49,6 +53,10 @@ export const ClientSideStateUpdate = ({
     if (viewSettings) {
       store.dispatch(setViewSettings(viewSettings))
     }
+    if (tokenPayload) {
+      store.dispatch(setTokenPayload(tokenPayload))
+      tokenPayload.internalUserId && store.dispatch(setInternalUserId(tokenPayload.internalUserId))
```

Comment:
Since we are already setting the `tokenPayload` in the store, why is it necessary to set `internalUserId` separately? I think the `tokenPayload` itself would be enough.